### PR TITLE
Make postprocessing compatible with nobuco

### PR DESCRIPTION
**Pull request by Pbatch**
There are 2 changes that allow the post-processing to be compatible with Nobuco (https://github.com/AlexanderLutsenko/nobuco), a PyTorch to Tensorflow converter.

They are as follows:
1) Use `dim` instead of `axis` for PyTorch functions.
2) Make sure that the labels have the same type as the boxes and scores (so that when they're concatenated everything has the same dtype).

With these changes I was able to run YOLOv10 in my web/mobile app CameraChess (https://www.camerachess.com)

Thank you for your contributions to the field :)

### CODE REVIEW

**Comment by jameslahm** [comment]
Thank you!

File: ultralytics/utils/ops.py [hunk 1]
```diff
@@ -161,20 +161,20 @@ def nms_rotated(boxes, scores, threshold=0.45):
...
+        agnostic=False,
+        multi_label=False,
+        labels=(),
+        max_det=300,
+        nc=0,  # number of classes (optional)
+        max_time_img=0.05,
+        max_nms=30000,
+        max_wh=7680,
+        in_place=True,
+        rotated=False,
```
Comment (jameslahm) [inline ultralytics/utils/ops.py:177]: Could you please revert this change? Thanks!

File: ultralytics/utils/ops.py [hunk 2]
```diff
@@ -560,7 +560,7 @@ def xywhr2xyxyxyxy(rboxes):
     cos, sin = (np.cos, np.sin) if is_numpy else (torch.cos, torch.sin)
 
     ctr = rboxes[..., :2]
-    w, h, angle = (rboxes[..., i : i + 1] for i in range(2, 5))
+    w, h, angle = (rboxes[..., i: i + 1] for i in range(2, 5))
```
Comment (jameslahm) [inline ultralytics/utils/ops.py:563]: Could you please also revert this change? Thanks!

File: ultralytics/utils/ops.py [hunk 3]
```diff
@@ -848,17 +848,18 @@ def clean_str(s):
...
-    max_scores, index = torch.topk(max_scores, max_det, axis=-1)
+    max_scores, index = torch.topk(max_scores, max_det, dim=-1)
     index = index.unsqueeze(-1)
     boxes = torch.gather(boxes, dim=1, index=index.repeat(1, 1, boxes.shape[-1]))
     scores = torch.gather(scores, dim=1, index=index.repeat(1, 1, scores.shape[-1]))
 
-    scores, index = torch.topk(scores.flatten(1), max_det, axis=-1)
-    labels = index % nc
+    scores, index = torch.topk(scores.flatten(1), max_det, dim=-1)
+    labels = (index % nc).to(torch.float32)
```
Comment (jameslahm) [inline ultralytics/utils/ops.py:862]: We previously added the cast for labels here
https://github.com/THU-MIG/yolov10/blob/13f6ab770d561e715a094f94594ca34b80dba1ee/ultralytics/nn/modules/head.py#L523 Could you please also revert this change? Thanks!

Comment (jameslahm) [inline ultralytics/utils/ops.py:862]: We previously added the cast for labels here
https://github.com/THU-MIG/yolov10/blob/13f6ab770d561e715a094f94594ca34b80dba1ee/ultralytics/nn/modules/head.py#L523 Could you please also revert this change? Thanks!

**Comment by Pbatch** [comment]
Sorry my IDE did some weird little changes. It's just the axis -> dim change now which should be all that's required!

**Comment by jameslahm** [comment]
Thank you!
